src/Apps/Components/AppShell.tsx

```diff
@@ -40,6 +42,10 @@ export const AppShell: React.FC<React.PropsWithChildren<AppShellProps>> = props
         logger.error(error)
       }
     }
+
+    // Preload current route's JS bundle
+    routeConfig?.onPreloadJS?.()
```

Comment:
Invoke the new function in the app shell (for when users first land on a specific page via url)